ipv8/attestation/communication_manager.py

```diff
@@ -189,7 +189,62 @@ def verify(self, peer: Peer, attribute_hash: bytes, reference_values: typing.Lis
                                                            self.on_verification_results, id_format)
 
 
-class CommunicationManager(object):
+class PseudonymFolderManager:
+    """
+    Perform file management of pseudonyms.
+    """
+
+    def __init__(self, directory: str) -> None:
+        """
+        Manage/store pseudonyms in a given directory.
+
+        :param directory: the directory to store pseudonyms
+        :returns: None
+        """
+        self.pseudonym_folder = directory
+        self.crypto = ECCrypto()
+
+    def get_private_key(self, name: str) -> PrivateKey:
```

Comment:
With `get_` in the name it gives the sense of query method but it is also creating the key if it does not exist. Therefore, I suggest to rename as `get_or_create_private_key`.